extension/lib/doc.js

```diff
@@ -0,0 +1,119 @@
+"use strict";
+(function(){
+var namespace = "lib/doc.js";
+if (window[namespace] === true) {
+    return;
+} else {
+    window[namespace] = true;
+}
+ 
+/*
+Class dedicated to keeping track of model of document. 
+Note: Called "Doc" because Document is already used in js. 
+*/
+
+class Doc {
+    constructor(readableDomIds) {
+        this.readableDomIds = readableDomIds; // string[]; List of all container IDs
+        this.containerId = null; // str; ID of current container
+        this.dict = null; // { str : int } , Set in calcTotalWords function
```

Comment:
if too tiring to refactor, im fine if u leave it like this, but...
consider choosing var name that describes the fcn, rather than just the type
some choices:
- wordTally
- wordCount

if u were to also include the type as part of the name im cool w/ that too.
E.g.
- wordCountDict
But i think the function of the variable is more important.

Reply:
Oh oops I was literally thinking of a dictionary in the physical sense (like a book that contains words), I agree this should for sure be changed

Reply:
Either way a poor name